framework/include/actions/Action.h

```diff
@@ -91,15 +116,18 @@ class Action : public ConsoleStreamInterface
   /// Input parameters for the action
   InputParameters _pars;
 
-  /// The name of the action
-  std::string _name;
-
-  /// The short name of the action
-  std::string _short_name;
-
   // The registered syntax for this block if any
   std::string _registered_identifier;
 
+  /// The complete syntax of the action
+  const std::string & _full_syntax;
+
+  /// The base syntax of the action
+  std::string _base_syntax;
+
+  /// The name of the action
+  std::string _name;
```

Comment:
Not sure if these new variable names are any better/worse than the old ones. I don't like syntax since they really don't have to be syntax... Let's talk about better names.